### PR TITLE
Set response status code in transaction "response" context.

### LINKED ISSUE #2289: Add response status code to transaction response context
### Problem Statement

Make sure we save the status code in `contexts.response.status_code` for transactions.

See also
* https://github.com/getsentry/sentry-python/issues/2099
* https://github.com/getsentry/team-sdks/issues/9

### Solution Brainstorm

Do it.

### PR DESCRIPTION
**Pull request by antonpirker**
Make sure that the HTTP response status code be set in the transactions "response" context. 

This works in WSGI (was already calling `set_http_status`.) Also added this to ASGI projects.

Fixes #2289

### CODE REVIEW

File: sentry_sdk/integrations/asgi.py [hunk 1]
```diff
@@ -178,11 +182,27 @@ async def _run_app(self, scope, callback):
...
-                        # XXX: Would be cool to have correct span status, but we
-                        # would have to wrap send(). That is a bit hard to do with
-                        # the current abstraction over ASGI 2/3.
                         try:
-                            return await callback()
+
+                            async def _sentry_wrapped_send(event):
+                                # type: (Dict[str, Any]) -> Any
+                                is_http_response = (
+                                    event["type"] == "http.response.start"
```
Comment (sentrivana) [inline sentry_sdk/integrations/asgi.py:190]: Is `event["type"]` guaranteed to always be there for any event? Would it make sense to `event.get("type")` instead?

File: sentry_sdk/integrations/asgi.py [hunk 2]
```diff
@@ -132,20 +132,24 @@ def _run_asgi2(self, scope):
...
+    async def _run_app(self, scope, receive, send, asgi_version):
+        # type: (Any, Any, Any, Any, int) -> Any
         is_recursive_asgi_middleware = _asgi_middleware_applied.get(False)
         if is_recursive_asgi_middleware:
             try:
-                return await callback()
+                if asgi_version == 3:
+                    return await self.app(scope, receive, send)
+                else:
+                    return await self.app(scope)(receive, send)
```
[the remark this answers is not in the document]
Comment (antonpirker) [reply sentry_sdk/integrations/asgi.py:151]: You are right. Switch them around.

File: sentry_sdk/integrations/asgi.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/asgi.py:190]: done. better save then sorry!

File: sentry_sdk/integrations/asgi.py [hunk 3]
```diff
@@ -178,11 +182,27 @@ async def _run_app(self, scope, callback):
...
-                            return await callback()
+
+                            async def _sentry_wrapped_send(event):
+                                # type: (Dict[str, Any]) -> Any
+                                is_http_response = (
+                                    event["type"] == "http.response.start"
+                                    and transaction is not None
+                                )
+                                if is_http_response:
+                                    transaction.set_http_status(event["status"])
```
[the remark this answers is not in the document]
Comment (antonpirker) [reply sentry_sdk/integrations/asgi.py:194]: good catch. updated.

**Pull request merged**